bwe/BandwidthEstimator.cpp

```diff
@@ -221,28 +312,37 @@ double BandwidthEstimator::analyseCongestion(double actualDelay, const uint32_t
             _congestion.dip.bandwidthFloorKbps = 0;
         }
     }
+
     return congestionScale;
 }
 
 // in kbps
 double BandwidthEstimator::getEstimate(uint64_t timestamp) const
 {
-    const double estimatedBandwidth = std::min(_state(1), _congestion.dip.bandwidthCapKbps);
+    double estimatedBandwidth = std::min(_state(Bandwidth), _congestion.dip.bandwidthCapKbps);
+    if (_congestion.consecutiveOver < 50)
+    {
+        estimatedBandwidth =
+            math::clamp(_state(Bandwidth), _congestion.estimateBeforeCongestion, _congestion.dip.bandwidthCapKbps);
+    }
 
     if (_previousReceiveTime != 0 &&
         utils::Time::diffGT(_previousReceiveTime, timestamp, _config.silence.timeoutMs * utils::Time::ms))
     {
-        return std::max(_config.estimate.minReportedKbps,
-            std::min(_config.silence.maxBandwidthKbps, estimatedBandwidth * (1.0 - _config.silence.backOff)));
+        return math::clamp(estimatedBandwidth * (1.0 - _config.silence.backOff),
+            _config.estimate.minReportedKbps,
+            _config.silence.maxBandwidthKbps);
     }
-    return std::max(std::max(_congestion.dip.bandwidthFloorKbps, _config.estimate.minReportedKbps),
-        estimatedBandwidth * (1.0 - _congestion.margin));
+
+    return std::max({_congestion.dip.bandwidthFloorKbps,
+        _config.estimate.minReportedKbps,
+        estimatedBandwidth * (1.0 - _congestion.margin)});
```

Comment:
Congestion margin is used for severe choking and we try to cause sender to allow for link to drain in less than 10s.
Otherwise, estimator will continue to report full bandwidth and delay will persist for long time.